### ISSUE TITLE
Window was cut at the edge

### ISSUE THREAD

**Issue by tantalate** [awesomeWM/awesome#1118]
Output of `awesome --version`:

awesome v3.5.9 (Mighty Ravendard)
Build: Mar 12 2016 01:11:40 for x86_64 by gcc version 5.3.0 (builduser@rw)
Compiled against Lua 5.3.2 (running with Lua 5.3)
D-Bus support: √

I installed Archlinux in VMware on Mac OSX, and installed awsome as the WM.
However the windows are not displayed fully at the edge, like in the pic.
![window](http://7xrpnw.com1.z0.glb.clouddn.com/arch.png)

I have tried with xterm, urxvt etc, with -b -w options, and they have the same problem.

These worked fine in openbox.

**Comment by psychon** [comment]
Could you get the exact window position by running `xwininfo` in a terminal and clicking on one of these window? Are they really offscreen?
Also, the output of the following might be interesting (preferably when you only have a single client visible which is partly offscreen):
```
echo 'local c = client.get()[1] ; return require("gears.debug").dump_return({ geom=screen[1].geometry, warea=screen[1].workarea, bw=c.border_width, c_geom=c:geometry()})' | awesome-client
```

**Comment by leetom** [comment]
`xwininfo` shown as in the figure.
![awesome2](https://cloud.githubusercontent.com/assets/1206216/18838594/974084b6-843a-11e6-90fc-b10cbee2b7e3.png)

and the other command as following.

![awesome](https://cloud.githubusercontent.com/assets/1206216/18838629/b66a301c-843a-11e6-9e8d-b0d57dc83454.png)

**Comment by psychon** [comment]
Thanks. Uhm... you can move the windows back to the visible space normally, right? (e.g. by actually moving the windows or by just having switching to a something else than the floating layout via mod+space)
So this issue is actually about "the initial position for all windows is offscreen"? Is it really all windows, or is it just your terminal? Which terminal is this?
What does your `rc.lua` look like? Is it just the default config or did you change something?
Do you know where messages from awesome end up? Something like `ls -l /proc/$(pidof awesome)/fd/1` might be able to tell you. It could be something like `~/.xsession-errors` or `/var/log/gdm.log`. If you know this, then doing something like this might be interesting (and then getting the log output which contains all the moves of the window and (hopefully) also why it was moved):
```
echo 'function log_position(c) local g = c:geometry() print(string.format("%dx%d+%d+%d for %s", g.width, g.height, g.x, g.y, c.name)) print(debug.traceback()) end ; client.connect_signal("manage", log_position) ; client.connect_signal("property::geometry", log_position)' | awesome-client
Now open a new window that will be offscreen
echo 'client.disconnect_signal("manage", log_position) ; client.disconnect_signal("property::geometry", log_position)' | awesome-client
```

**Comment by leetom** [comment]
Hi, it turns out that it only happens when I first start the terminal (xterm or urxvt). After I pressed Mod+Space, the scrollbar appeared and won't disappear even I switch to the original layout. 

I have tried to start chromium and it looks normal.

`ls -l /proc/$(pidof awesome)/fd/1` shows the log file is /dev/tty1 and the result of the two commands is as following:

![qq20160927-0](https://cloud.githubusercontent.com/assets/1206216/18842643/4e3051ee-8448-11e6-9a55-bcef2176ba38.png)

I haven't touched rc.lua except for the Modkey.


The problem becomes not a big deal after I modified the monkey.

**Comment by Elv13** [comment]
Does this still happens in v4.1?

**Comment by psychon** [comment]
Actually, I would be more interested in "does this still happen with git master or v4.2 once it is released?". This might have been fixed with 1332288f9de1affaaad8d292816cdf1b17e645e7.

**Comment by Elv13** [comment]
Assumed to be fixed
